File: backend/src/main/java/com/lms_erp/program/service/ProgramService.java
package com.lms_erp.program.service;

import com.lms_erp.program.dto.*;

import java.util.List;

public interface ProgramService {

    ProgramResponse createProgram(
            ProgramCreateRequest request
    );

    List<ProgramResponse> getAllPrograms();

    ProgramResponse getProgramById(
            Long programId
    );

    ProgramResponse updateProgram(
            Long programId,
            ProgramUpdateRequest request
    );

    void deleteProgram(
            Long programId
    );

    List<ProgramResponse> getActivePrograms();

    List<ProgramResponse> getInactivePrograms();
}

File: lms-frontend/src/program/services/programService.js
import api from "../../api/axios";

export const createProgram = (data) => {
    return api.post("/api/programs", data);
};

export const getAllPrograms = () => {
    return api.get("/api/programs");
};

export const getActivePrograms = () => {
    return api.get("/api/programs/active");
};

export const getInactivePrograms = () => {
    return api.get("/api/programs/inactive");
};

export const getProgramById = (id) => {
    return api.get(`/api/programs/${id}`);
};

export const updateProgram = (id, data) => {
    return api.put(`/api/programs/${id}`, data);
};

export const deleteProgram = (id) => {
    return api.delete(`/api/programs/${id}`);
};